### DISCUSSION TITLE
Why can I not match a sentence with comma

### DISCUSSION THREAD

**Discussion by trsh** [nlp-compromise/nlp_compromise#980] [Q&A]
```typescript
let doc = nlp(`You can eat, sugar and water`)
    let m = doc.match('You can eat, [*]')
    console.log((<View>m.groups('0')).out('array'));
```

If I remove `,` in sentence and match, all works, but I need it.

**Update:**

```typescript
let doc = nlp(`You can eat sugar, cold stuff and water`)
    let m = doc.match('You can eat [*], [*] and [*]')
    console.log((<View>m.groups('0')).out('array'));
    console.log((<View>m.groups('1')).out('array'));
    console.log((<View>m.groups('2')).out('array'));
```
I want to extract `sugar`, `cold stuff`, `water`. But I can't. When I use comma in match, nothing is matched. If I dont use it, the results are incorrect.

**Update2:**

Another example:

```typescript
let doc = nlp(`Eat sugar! and melon`);
    let m = doc.match('Eat [*] and [*]')
    console.log((<View>m.groups('0')).out('array'));
    console.log((<View>m.groups('1')).out('array'));
```
It doesn't work, unless I remove "!" from text. Why?

**Comment by spencermountain** [comment]
hey Janis, good questions. 
yeah - the match statements are not meant to be natural language in, with punctuation and stuff. That's an interesting thought - how come you need the comma in the match statement? 

`[*]` is greedy, and i would recommend `[.]` to match one term, instead.

In the last one - with the exclamation, it is splitting into two sentences. It is not possible to match between two sentences, but that would be a neat feature to add.
cheers

**Reply by trsh** [reply]
To answer you question look at my example: "You can eat sugar, cold stuff and water", where I want to extract `sugar`, `cold stuff` and `water`. To do this correctly I must take everything with `and` and `,` in between, after `You can eat`. And that is the reason. 
However I went for a splitting strategy with regexes.
